src/Components/Login.js

```diff
@@ -0,0 +1,129 @@
+import React, { useState } from 'react';
+import Box from '@mui/material/Box'
+import TextField from '@mui/material/TextField'
+import Button from '@mui/material/Button'
+import Card from './Card.js'
+import './Login.scss'
+
+const Login = () => {
+
+    const [isUsernameError, setIsUsernameError] = useState(false)
+    const [isPassword1Error, setIsPassword1Error] = useState(false);
+    const [isPassword2Error, setIsPassword2Error] = useState(false);
+    const [isEmailError, setIsEmailError] = useState(false);
+    const [enteredPassword1, setEnteredPassword1] = useState('');
+    const [enteredPassword2, setEnteredPassword2] = useState('');
+    const [enteredEmail, setEnteredEmail] = useState('');
+    const [enteredUsername, setEnteredUsername] = useState('');
+
+    const usernameChangehandler = (event) => {
+        setEnteredUsername(event.target.value);
+    }
+
+    const password1ChangeHandler = (event) => {
+        setEnteredPassword1(event.target.value)
+    }
+
+    const password2ChangeHandler = (event) => {
+        setEnteredPassword2(event.target.value)
+    }
+    
+    const emailChangeHandler = (event) => {
+        setEnteredEmail(event.target.value)
+    }
+    
+    const validateEmailHandler = () => {
+
+        if(enteredEmail.trim().length==0 || !enteredEmail.includes('@')){
+            setIsEmailError(true);
+        } else {
+            setIsEmailError(false);
+        }
+    }
+
+    const validateUsernameHandler = () => {
+        if(enteredUsername.trim().length==0 || enteredUsername.trim().length<6){
+            setIsUsernameError(true)
+        } else {
+            setIsUsernameError(false);
+        }
+    }
+
+    const validatePasswordMatchHandler = () => {
+        if(enteredPassword1 != enteredPassword2){
+            setIsPassword2Error(true);
```

Comment:
Much more concise way is 
`setIsPassword2Error(enteredPassword1 !== enteredPassword2)` since the expression in the `if` clause is an equality check. Which returns a `boolean` that can be assigned to our state

Reply:
Done